aten/src/ATen/native/TensorShape.cpp

```diff
@@ -898,20 +898,38 @@ Tensor numpy_T(const Tensor &self) {
 
 Tensor view(const Tensor& self, IntArrayRef size) {
   auto inferred_size = at::infer_size(size, self.numel());
-  if (self.sizes() == inferred_size) {
-    return self;
-  }
-
   auto stride = at::detail::computeStride(self.sizes(),
-                                           self.strides(),
-                                           inferred_size);
+                                          self.strides(),
+                                          inferred_size);
   TORCH_CHECK(stride.has_value(), "view size is "
     "not compatible with input tensor's size and stride (at least one dimension"
     " spans across two contiguous subspaces). Use .reshape(...) instead.");
   auto stride_value = *stride;
-  auto self_ = self.clone();
-  self_.set_(self.storage(), self.storage_offset(), inferred_size,
-             stride_value);
+  auto self_ = self.alias();
+  self_.set_(
+    self.storage(), self.storage_offset(), inferred_size, stride_value);
+  return self_;
+}
+
+Tensor alias(const Tensor& self) {
+  Tensor self_;
+  if (self.is_quantized()) {
+    TORCH_CHECK(self.qscheme() == kPerTensorAffine,
```

Comment:
Should we allow `kPerTensorSymmetric` in here as well?

Reply:
we don't support that right now

Reply:
nit: shouldn't the QTensorImpl constructor check that then?

Reply:
I think we have per channel quantizer right now. looking again, the following code actually works for whatever tensor we have, so we can remove this check.